app/lib/actions/pulp/abstract_async_task.rb

```diff
@@ -79,7 +79,7 @@ def external_task
         output[:pulp_tasks]
       end
 
-      def cancel!
+      def cancel!(raise_error = true)
```

Comment:
From a reading the code perspective, it is quite confusing (as this conversation indicates). Two methods could work to alleviate that. Alternatively, we could rescue the exception and if 'raise_error' is true, we re-raise the exception otherwise we eat it. Another option would be to pass in the error message that we want to have instead of have it use that?